src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -203,8 +210,71 @@ def wrap(*args, **kwargs):
                             )
                             raise ne.with_traceback(tb) from None
                         else:
+                            from snowflake.snowpark._internal.analyzer.select_statement import (
+                                Selectable,
+                            )
+
+                            # We need the potential double quotes for invalid identifier
+                            match = SnowflakePlan.Decorator.__wrap_exception_regex_match_with_double_quotes.match(
+                                e.msg
+                            )
+                            if not match:  # pragma: no cover
+                                ne = SnowparkClientExceptionMessages.SQL_EXCEPTION_FROM_PROGRAMMING_ERROR(
+                                    e
+                                )
+                                raise ne.with_traceback(tb) from None
+                            col = match.group(1)
+
+                            quoted_identifiers = []
+                            for child in children:
+                                plan_nodes = child.children_plan_nodes
+                                for node in plan_nodes:
+                                    if isinstance(node, Selectable):
+                                        quoted_identifiers.extend(
+                                            node.snowflake_plan.quoted_identifiers
+                                        )
+                                    else:
+                                        quoted_identifiers.extend(
+                                            node.quoted_identifiers
+                                        )
+
+                            def add_single_quote(string: str) -> str:
+                                return f"'{string}'"
+
+                            # We can't display all column identifiers in the error message
+                            if len(quoted_identifiers) > 10:
+                                quoted_identifiers_str = f"[{', '.join(add_single_quote(q) for q in quoted_identifiers[:10])}, ...]"
```

Comment:
curious why the value is set to 10 columns?

Reply:
I had similar thoughts. seems like identifier size is upper-bounded by 255: https://docs.snowflake.com/en/sql-reference/identifiers-syntax while there is no upper bound on number of cols so I agree with what jianzhun has done here. What is we expand the message to say something like "use df.columns to list all available columns for the dataframe".

Reply:
Yeah it's just to avoid display a very long error messages if there are a lot of columns